js/test/Ice/properties/Client.ts

```diff
@@ -99,12 +108,40 @@ export class Client extends TestHelper {
         }
 
         {
-            out.write("testing property regexp pattern...");
+            out.write("testing that setting an unknown ice property throws an exception...");
             const properties = Ice.createProperties();
-            properties.setProperty("Ice.Default.Locator.Context.Foo", "Bar");
-            const value = properties.getProperty("Ice.Default.Locator.Context.Foo");
-            test(value == "Bar");
+            try {
+                properties.setProperty("Ice.UnknownProperty", "bar");
+                test(false);
+            } catch (ex) {
+                test(ex.message == "unknown Ice property: Ice.UnknownProperty");
+            }
+            out.writeLine("ok");
+        }
+
+        {
+            const communicator = Ice.initialize(args);
+            const properties = communicator.getProperties();
+
+            out.write("testing that creating an object adapter with unknown properties throws an exception...");
+            properties.setProperty("FooOA.Endpoints", "tcp -h 127.0.0.1");
```

Comment:
We should fix this test, once we remove `Endpoints` from the valid OA JavaScript properties. We already have a TODO in PropertyNames.xml.